WinImageBuilder.psm1

```diff
@@ -1195,182 +1045,93 @@ function New-WindowsOnlineImage {
-     In case the hardware on which the image is generated will also be the hardware on
-     which the image will be deployed this can be set to true, otherwise the spawned
-     instance is prone to BSOD.
-    .PARAMETER InstallMaaSHooks
-     If set to true, MaaSHooks will be installed.
-    .PARAMETER VirtIOBasePath
-     The drive letter of the mounted VirtIO drivers ISO file.
-    .PARAMETER PurgeUpdates
-     If set to true, will run DISM with /resetbase option. This will reduce the size of 
-     WinSXS folder, but after that Windows updates cannot be uninstalled.
-    .PARAMETER DisableSwap
-     DisableSwap option will disable the swap when the image is generated and will add a setting
-     in the Unattend.xml file which will enable swap at boot time during specialize step. This
-     is required as by default, the amount of swap space on Windows machine is directly
-     proportional to the RAM size and if the image has in the initial stage low disk space,
-     the first boot will fail due to not enough disk space. The swap is set to the default
-     automatic setting right after the resize of the partitions is performed by cloudbase-init.
-    #>
     [CmdletBinding()]
-    Param(
-        [parameter(Mandatory=$true, ValueFromPipeline=$true)]
-        [string]$WimFilePath = "D:\Sources\install.wim",
-        [parameter(Mandatory=$true)]
-        [string]$ImageName,
-        [parameter(Mandatory=$true)]
-        [string]$VirtualDiskPath,
-        [parameter(Mandatory=$true)]
-        [Uint64]$SizeBytes,
-        [ValidatePattern("^$|^\S{5}-\S{5}-\S{5}-\S{5}-\S{5}")]
-        [parameter(Mandatory=$false)]
-        [string]$ProductKey,
-        [parameter(Mandatory=$false)]
-        [ValidateSet("VHD", "VHDX", "QCow2", "VMDK", "RAW", ignorecase=$false)]
-        [string]$VirtualDiskFormat = "VHDX",
-        [ValidateSet("BIOS", "UEFI", ignorecase=$false)]
-        [string]$DiskLayout = "BIOS",
-        [parameter(Mandatory=$false)]
-        [string]$VirtIOISOPath,
-        [parameter(Mandatory=$false)]
-        [array]$ExtraFeatures = @(),
-        [parameter(Mandatory=$false)]
-        [string]$ExtraDriversPath,
-        [parameter(Mandatory=$false)]
-        [switch]$InstallUpdates,
-        [parameter(Mandatory=$false)]
-        [string]$AdministratorPassword = "Pa`$`$w0rd",
-        [parameter(Mandatory=$false)]
-        [string]$UnattendXmlPath = "$scriptPath\UnattendTemplate.xml",
-        [parameter(Mandatory=$false)]
-        [switch]$PersistDriverInstall = $true,
-        [parameter(Mandatory=$false)]
-        [switch]$InstallMaaSHooks,
-        [parameter(Mandatory=$false)]
-        [string]$VirtIOBasePath,
-        [parameter(Mandatory=$false)]
-        [switch]$PurgeUpdates,
-        [parameter(Mandatory=$false)]
-        [switch]$DisableSwap,
-        [parameter(Mandatory=$false)]
-        [string]$ZipPassword
-    )
+    param
+    (
+        [parameter(Mandatory=$false, ValueFromPipeline=$true)]

```

Comment:
these two params and the 1059 logic should be also set for the other generation methods